components/cdc/src/initializer.rs

```diff
@@ -207,12 +204,24 @@ impl<E: KvEngine> Initializer<E> {
             Err(box_err!("scan canceled"))
         };
 
-        if self.downstream_state.load() == DownstreamState::Stopped {
-            return on_cancel();
-        }
+        let scan_concurrency_semaphore = self.scan_concurrency_semaphore.clone();
+        let mut downstream_closed = Box::new(future::pending().shared())
+            as Box<dyn Future<Output = ()> + Send + Sync + Unpin>;
+        swap(&mut self.downstream_closed, &mut downstream_closed);
+        let _permit = select! {
+            _ = (&mut downstream_closed).fuse() => {
+                return on_cancel();
+            }
+            x = scan_concurrency_semaphore.acquire().fuse() => match x {
```

Comment:
if block on `scan_concurrency_semaphore` here, will CDC client detect the block and stop subscribe outside?

Reply:
Yes, the branch is in a `select` block.